File: src/cache/openrs/util/FileChannelUtils.java
package cache.openrs.util;

import java.io.EOFException;
import java.io.IOException;
import java.nio.ByteBuffer;
import java.nio.channels.FileChannel;


public final class FileChannelUtils {

    
    public static void readFully(FileChannel channel, ByteBuffer buffer, long ptr) throws IOException {
        while (buffer.remaining() > 0) {
            long read = channel.read(buffer, ptr);
            if (read == -1L) {
                throw new EOFException();
            } else {
                ptr += read;
            }
        }
    }

    
    private FileChannelUtils() {

    }

}

File: src/cache/alex/util/FileChannelUtils.kt
package cache.alex.util

import java.io.EOFException
import java.io.IOException
import java.nio.ByteBuffer
import java.nio.channels.FileChannel


object FileChannelUtils {
    
    @Throws(IOException::class)
    fun readFully(channel: FileChannel, buffer: ByteBuffer, ptr: Long) {
        var ptr = ptr
        while (buffer.remaining() > 0) {
            val read = channel.read(buffer, ptr).toLong()
            if (read == -1L) {
                throw EOFException()
            } else {
                ptr += read
            }
        }
    }
}
